File: src/lambda/dms-switch-py/utils/get_dms_replication_status.py

def get_dms_replication_status(dms, replication_config_arn):
    filters = [
        {
            'Name': 'replication-config-arn',
            'Values': [replication_config_arn]
        }
    ]
    response = dms.describe_replications(Filters=filters)
    status = response['Replications'][0]['Status'] if response.get(
        'Replications') else None
    return status

File: src/lambda/dms-switch/utils/get-dms-replication-status.ts
import {
  DatabaseMigrationServiceClient,
  DescribeReplicationsCommand,
} from '@aws-sdk/client-database-migration-service';

export const getDmsReplicationStatus = async ({
  dms,
  ReplicationConfigArn: replicationConfigArn,
}: {
  dms: DatabaseMigrationServiceClient;
  ReplicationConfigArn: string;
}): Promise<string> => {
  const configDescriptionCmd = new DescribeReplicationsCommand({
    Filters: [
      {
        Name: 'replication-config-arn',
        Values: [replicationConfigArn],
      },
    ],
  });
  const taskDescription = await dms.send(configDescriptionCmd);
  
  const status = `${taskDescription?.Replications?.[0].Status}`;
  return status;
};